applications/api-cat/src/test/java/no/acat/model/HarvestMetadataFactoryTest.java

```diff
@@ -0,0 +1,78 @@
+package no.acat.model;
+
+import no.dcat.shared.HarvestMetadata;
+import no.dcat.shared.testcategories.UnitTest;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+@RunWith(PowerMockRunner.class)
+@Category(UnitTest.class)
+public class HarvestMetadataFactoryTest {
+
+  Date harvestDate;
+  HarvestMetadata harvestMetadata;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    harvestDate = new DateTime(2018, 6, 20, 0, 0).toDate();
+    harvestMetadata = new HarvestMetadata();
+    harvestMetadata.setFirstHarvested(harvestDate);
+    harvestMetadata.setLastChanged(harvestDate);
+    harvestMetadata.setLastHarvested(harvestDate);
+  }
+
+  @PrepareForTest({HarvestMetadataFactory.class})
+  @Test
+  public void test_if_has_changed() throws Exception {
+
+    PowerMockito.mockStatic(HarvestMetadataFactory.class);
+    PowerMockito.doCallRealMethod()
+        .when(
+            HarvestMetadataFactory.class,
+            "recordHarvest",
+            Mockito.any(HarvestMetadata.class),
+            Mockito.any(Date.class),
+            Mockito.any(boolean.class));
+
+    HarvestMetadata metadata =
+        HarvestMetadataFactory.recordHarvest(harvestMetadata, harvestDate, true);
+    Assert.assertTrue(metadata.getChanged().get(0).equals(harvestDate));
+  }
+
+  @PrepareForTest({HarvestMetadataFactory.class})
+  @Test
+  public void test_if_not_changed() throws Exception {
```

Comment:
jeg er ikke helt sikkert hava vil du teste her, og beskrivelse sier det ikke.

Hvis en av de tester var at change er registrert på en riktig måte, da ten andre test skulle være enklere basisfunksjonalitet, og skulle være posisjonert først.
for eksempel, kanskje "initializeHarvestRecordIfMissing" og "changeLastHarvestedIfExisting"

Reply:
Ikke helt sikker hva du skrev